prereise/gather/demanddata/bldg_electrification/load_projection.py

```diff
@@ -0,0 +1,674 @@
+        hours_utc_year = pd.date_range(
+            start=f"{year}-01-01", end=f"{year+1}-01-01", freq="H", tz="UTC"
+        )[:-1]
+        temp_df_yr, stats = zonal_data(puma_data_zone, hours_utc_year, year)
+        temp_df = pd.concat([temp_df, temp_df_yr], ignore_index=True)
+    temp_df.index = hours_utc_weather_years
+
+    # compute least-square estimator for relation between WBT and DBT
+    t_bpc_start = 10
+    db_wb_regr_df = temp_df[
+        (temp_df.index.year == base_year) & (temp_df["temp_c"] >= t_bpc_start)
+    ]
+    db_wb_fit = np.polyfit(db_wb_regr_df["temp_c"], db_wb_regr_df["temp_c_wb"], 2)
+
+    hourly_fits_df = pd.read_csv(
+        f"https://besciences.blob.core.windows.net/datasets/bldg_el/dayhour_fits/{zone_name}_dayhour_fits_{base_year}.csv",
+        index_col=0,
+    )
+
+    midperfhp_cop = pd.read_csv("./data/cop_temp_htg_midperfhp.csv")
+    advperfhp_cop = pd.read_csv("./data/cop_temp_htg_advperfhp.csv")
+    midperfhp_cop.index = midperfhp_cop["temp"]
+    advperfhp_cop.index = advperfhp_cop["temp"]
+
+    if base_scen.hp_type_heat == "midperfhp":
+        base_hp_heat_cop = midperfhp_cop
+    elif base_scen.hp_type_heat == "advperfhp":
+        base_hp_heat_cop = advperfhp_cop
+    else:
+        raise KeyError("hp type not defined. Choose from 'midperfhp' and 'advperfhp'")
+
+    # run reference scenario
+    zone_profile_refload_MWh = pd.DataFrame(  # noqa: N806
+        {"hour_utc": hours_utc_weather_years}
+    )
+
+    energy_list = zone_profile_refload_MWh.hour_utc.apply(
+        lambda x: temp_to_energy(
+            temp_df.loc[x], hourly_fits_df, db_wb_fit, base_scen, base_hp_heat_cop
+        )
+    )
+
+    (
+        zone_profile_refload_MWh["base_load_mw"],
+        zone_profile_refload_MWh["heat_hp_load_mw"],
+        zone_profile_refload_MWh["heat_resist_load_mw"],
+        zone_profile_refload_MWh["cool_load_mw"],
+    ) = (
+        energy_list.apply(lambda x: x[0]),
+        energy_list.apply(lambda x: x[1]),
+        energy_list.apply(lambda x: x[2]),
+        energy_list.apply(lambda x: x[3]),
+    )
+    zone_profile_refload_MWh.set_index("hour_utc", inplace=True)
+
+    # scale energy to each projection scenarios
+    elec_profile_load_mwh = {}
+    zone_profile_load_mwh = {}
+    zone_profile_load_mwh["base"] = zone_profile_refload_MWh
+    ff2hp_profile_load_mwh = {}
+    for id, scenario in new_scens.items():
+        elec_profile_load_mwh[id] = scale_energy(
+            zone_profile_refload_MWh,
+            temp_df,
+            base_scen,
+            scenario,
+            midperfhp_cop,
+            advperfhp_cop,
+        )
+        ff2hp_profile_load_mwh[id] = ff_electrify_profiles(
+            weather_years, puma_data_zone, scenario, base_scen
+        )
+        zone_profile_load_mwh[id] = pd.concat(
+            [elec_profile_load_mwh[id], ff2hp_profile_load_mwh[id]], axis=1
+        )
+
+    return zone_profile_load_mwh
+
+
+if __name__ == "__main__":
```

Comment:
Instead of calling `load_projection.py` as a script, we can get rid of this block and put the code below in a demonstration notebook that generates the projected load profile based on the sample inputs.

Reply:
Can we do this on our next pull request? This is a good suggestion. And I'm considering putting together a notebook script that allow the users to create scenario input files and then in the same notebook, they can run load_projection.py.

Reply:
Sure. Let's put this refactor in the follow-up PR then.